test/integration/implptr/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+
+add_library(pimpl SHARED classes.cc)
```

Comment:
Small suggestion, name this library something like `implptr_testing` so it's clearer what it's for and it matches the implementation.

I think this is what's failing to link on Windows CI:

```
[30.141s] LINK : fatal error LNK1181: cannot open input file '..\..\..\lib\Release\pimpl.lib' [C:\Jenkins\workspace\ignition_utils-ci-pr_any-windows7-amd64\ws\build\ignition-utils0\test\integration\implptr\INTEGRATION_ImplPtr_TEST.vcxproj]
```

https://build.osrfoundation.org/job/ignition_utils-ci-pr_any-windows7-amd64/3/console